magicctapipe/io/io.py

```diff
@@ -908,28 +737,41 @@ def load_mc_dl2_data_file(input_file, quality_cuts, event_type, weight_type_dl2)
         If the input event type is not known
     """
 
+    TEL_NAMES, TEL_COMBINATIONS = telescope_combinations(config)
+    combo_types = np.asarray(range(len(TEL_COMBINATIONS)))
+    three_or_more = []
+    for n, combination in enumerate(TEL_COMBINATIONS.values()):
+        if len(combination) >= 3:
+            three_or_more.append(n)
+
     # Load the input file
     df_events = pd.read_hdf(input_file, key="events/parameters")
     df_events.set_index(["obs_id", "event_id", "tel_id"], inplace=True)
     df_events.sort_index(inplace=True)
 
-    df_events = get_stereo_events_old(df_events, quality_cuts)
+    df_events = get_stereo_events(
+        df_events, config, quality_cuts, eval_multi_combo=False
+    )
 
     logger.info(f"\nExtracting the events of the '{event_type}' type...")
 
     if event_type == "software":
         # The events of the MAGIC-stereo combination are excluded
-        df_events.query("(combo_type > 0) & (magic_stereo == True)", inplace=True)
+        df_events.query(
+            f"(combo_type < {combo_types[-1]}) & (magic_stereo == True)", inplace=True
```

Comment:
taking into account the other problem that we had with the combo_type it might be safer to access this combination by name "M1_M2"

Reply:
fixed